.github/workflows/cmake.yml

```diff
@@ -20,277 +19,247 @@ jobs:
+            cxxver: 17
+            msvcver: 2017
+          - name: "Windows MSVC 2017 (x64) C++20"
+            os: windows-2019
+            buildtype: Release
+            cxx: "cl"
+            environment_script: "C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Auxiliary/Build/vcvars64.bat"
+            cxxver: 20
+            msvcver: 2017
+          - name: "Windows MSVC 2019 (x64) C++17"
+            os: windows-2019
+            buildtype: Release
+            cxx: "cl"
+            environment_script: "C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Auxiliary/Build/vcvars64.bat"
+            cxxver: 17
+          - name: "Windows MSVC 2019 (x64) C++20"
+            os: windows-2019
+            buildtype: Release
+            cxx: "cl"
+            environment_script: "C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Auxiliary/Build/vcvars64.bat"
+            cxxver: 20
+          - name: "Windows MSVC 2022 (x64) C++17"
+            os: windows-2022
+            buildtype: Release
+            cxx: "cl"
+            environment_script: "C:/Program Files/Microsoft Visual Studio/2022/Enterprise/VC/Auxiliary/Build/vcvars64.bat"
+            cxxver: 17
+          - name: "Windows MSVC 2022 (x64) C++20"
+            os: windows-2022
+            buildtype: Release
+            cxx: "cl"
+            environment_script: "C:/Program Files/Microsoft Visual Studio/2022/Enterprise/VC/Auxiliary/Build/vcvars64.bat"
+            cxxver: 20
+          - name: "Msys2/CLANG"
+            os: windows-2019
+            buildtype: Release
+            mingw: MINGW64
+            mingw-arch: x86_64
+            cxx: "clang"
+            cxxver: 20
+          - name: "Msys2/GCC"
+            os: windows-2019
+            buildtype: Release
+            mingw: MINGW64
+            mingw-arch: x86_64
+            cxx: "gcc"
+            cxxver: 20
+          - name: "Msys2/debugCLANG"
+            os: windows-2019
+            buildtype: Debug
+            mingw: MINGW64
+            mingw-arch: x86_64
+            cxx: "clang"
+            cxxver: 17
+          - name: "Msys2/debugGCC"
+            os: windows-2019
+            buildtype: Debug
+            mingw: MINGW64
+            mingw-arch: x86_64
+            cxx: "gcc"
+            cxxver: 17
     steps:
     - uses: actions/checkout@v3
+    - name: "Msys2"
+      if: (startsWith(matrix.config.os, 'windows-2019') && contains(matrix.config.mingw, 'MINGW'))
+      uses: msys2/setup-msys2@v2
+      with:
+        msystem: ${{ matrix.config.mingw }}
+        install: git
+        update: true
 
     - name: Install Clang 10
       id: install_clang_10
-      if: startsWith(matrix.config.os, 'ubuntu') && ( matrix.config.cxx == 'clang++-10' )
+      if: (startsWith(matrix.config.os, 'ubuntu') && ( matrix.config.cxx == 'clang++-10' ))
```

Comment:
Sorry! 
Refactoring cmake.yml was because the github editor showed several false positive errors.

**main branch**
![ghEditor_fakeError](https://github.com/andreasbuhr/cppcoro/assets/6756180/83341427-e6d7-45d2-b497-fd469b42cdc4)